magenta-lib/src/main/scala/magenta/tasks/tasks.scala

```diff
@@ -13,19 +13,21 @@ import software.amazon.awssdk.core.internal.util.Mimetype
 import software.amazon.awssdk.core.sync.{RequestBody => AWSRequestBody}
 import software.amazon.awssdk.services.s3.S3Client
 import software.amazon.awssdk.services.s3.model.{GetObjectRequest, ObjectCannedACL, PutObjectRequest}
+import software.amazon.awssdk.services.sts.StsClient
 
 import scala.util.control.NonFatal
 
 case class S3Upload(
   region: Region,
   bucket: String,
   paths: Seq[(S3Location, String)],
+  resources: DeploymentResources,
```

Comment:
we possibly don't actually need to pass this here as we've gone down the route of changing the signature of `execute` - so all that's needed in this task is an artifactclient (for `resolveMappings` on line 97`)

Reply:
update: I've removed this and gone back to having an implicit artifactClient everywhere